packages/dropdown/src/react/__stories__/index.story.js

```diff
@@ -578,3 +580,167 @@ storiesOf('portal', module).add('position', _ => (
+      selectedLabel,
+      selectedValue,
+      setMenuPosition,
+      setOpen,
+      width
+    } = useDropdown({
+      forwardedRef,
+      menu,
+      value,
+      onChange,
+      onClick,
+      placeholder
+    })
+    return (
+      <Dropdown.Layout
+        disabled={disabled}
+        style={style}
+        className={className}
+        onKeyDown={handleKeyDown}
+        label={<Dropdown.Label label={label} />}
+        menu={
+          <DropdownContext.Provider value={selectedValue}>
+            <Dropdown.Menu
+              inNode={inNode}
+              isOpen={isOpen}
+              menu={menu}
+              menuPosition={menuPosition}
+              onClick={handleMenuChange}
+              onClose={() => {
+                setOpen(false)
+              }}
+              ref={menuRef}
+              width={width}
+            />
+          </DropdownContext.Provider>
+        }
+        subLabel={<Dropdown.SubLabel subLabel={subLabel} />}
+        button={
+          <Dropdown.Button
+            appearance={appearance}
+            disabled={disabled}
+            error={error}
+            isOpen={isOpen}
+            onClick={handleToggleOpen}
+            setMenuPosition={setMenuPosition}
+            size={size}
+            ref={combinedRef}
+            {...rest}
+          >
+            {icon}
+            <div style={{ height: '100%', position: 'relative', flex: 1 }}>
+              <Dropdown.Selected
+                appearance={appearance}
+                label={longestMenuItemState.label}
+                placeholder={placeholder}
+                selectedLabel={selectedLabel}
+                size={size}
+              />
+            </div>
+          </Dropdown.Button>
+        }
+      />
+    )
+  }
+)
+
+const DropdownWithDynamicIcon = () => {
+  const [selected, setSelected] = useState()
+  const values = {
+    channel: {
+      value: 'channel',
+      icon: <Icon.ChannelIcon style={{ marginRight: 8 }} />,
+      label: 'Channel'
+    },
+    analytics: {
+      value: 'analytics',
+      icon: <Icon.AnalyticsIcon style={{ marginRight: 8 }} />,
+      label: 'Analytics'
+    },
+    authorKit: {
+      value: 'authorKit',
+      icon: <Icon.AuthorKitIcon style={{ marginRight: 8 }} />,
+      label: 'Author Kit'
+    },
+    labs: {
+      value: 'labs',
+      icon: <Icon.LabsIcon style={{ marginRight: 8 }} />,
+      label: 'Labs'
+    }
+  }
+  const handleChange = (e, value) => {
+    setSelected(value)
+  }
+  return (
+    <DropdownWithIcon
+      icon={
+        values[selected] ? (
+          values[selected].icon
+        ) : (
+          <div style={{ width: 24, height: 24, marginRight: 8 }} />
+        )
+      }
+      onChange={handleChange}
+      menu={
+        <>
+          {Object.values(values).map(({ value, icon, label }) => (
+            <Dropdown.Item value={value} key={value} icon={icon}>
+              {label}
+            </Dropdown.Item>
+          ))}
+        </>
```

Comment:
So there's some code around menu.props I need to review so that we can pass an array instead of just a fragment to the menu prop on Dropdown. This has nothing to do wit the refactor and is instead a  preexisting bug.

Reply:
Agreed, array support would be nice.